validator/client/validator_attest.go

```diff
@@ -0,0 +1,112 @@
+package client
+
+import (
+	"context"
+
+	"fmt"
+
+	"github.com/prysmaticlabs/prysm/shared/params"
+
+	"github.com/opentracing/opentracing-go"
+	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
+)
+
+// AttestToBlockHead completes the validator client's attester responsibility at a given slot.
+// It fetches the latest beacon block head along with the latest canonical beacon state
+// information in order to sign the block and include information about the validator's
+// participation in voting on the block.
+func (v *validator) AttestToBlockHead(ctx context.Context, slot uint64) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "validator.AttestToBlockHead")
+	defer span.Finish()
+	// First the validator should construct attestation_data, an AttestationData
+	// object based upon the state at the assigned slot.
+	attData := &pbp2p.AttestationData{
+		Slot:                 slot,
+		ShardBlockRootHash32: params.BeaconConfig().ZeroHash[:], // Stub for Phase 0.
+	}
+	req := &pb.CrosslinkCommitteeRequest{
+		Slot: slot,
+	}
+	resp, err := v.attesterClient.CrosslinkCommitteesAtSlot(ctx, req)
+	if err != nil {
+		log.Errorf("Could not fetch crosslink committees at slot %d: %v", slot, err)
+		return
+	}
```

Comment:
Could you add a check for `len(resp.Committee)` , so that if it is a zero-length committe we exit this and log an error